File: PHP/MinimumCostToMakeArrayEqualTest.php
<?php
/**
 * @version PHP 8.2.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit MinimumCostToMakeArrayEqualTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("MinimumCostToMakeArrayEqual.php");

class MinimumCostToMakeArrayEqualTest extends TestCase {

    public function testMinimumCostToMakeArrayEqual() {
        $obj = new MinimumCostToMakeArrayEqual();
        //Test 1
        $want = 8;
        $got = $obj->minCost( [1,3,5,2], [2,3,1,14] );
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = 0;
        $got = $obj->minCost( [2,2,2,2,2], [4,2,8,1,3] );
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }
}

File: GO/MinimumCostToMakeArrayEqual_test.go
/**
 * @version GO 1.20.5
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestMinimumCostToMakeArrayEqual
 */
package GO

import (
	"reflect"
	"testing"
)

func TestMinimumCostToMakeArrayEqual(t *testing.T) {
	tables := []struct {
		want int64
		nums []int
		cost []int
	}{
		{8, []int{1, 3, 5, 2}, []int{2, 3, 1, 14}},
		{0, []int{2, 2, 2, 2, 2}, []int{4, 2, 8, 1, 3}},
	}

	for _, table := range tables {
		got := minCost(table.nums, table.cost)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %d but the return was this: %d", table.want, got)
		}
	}
}
